ios-xcframework/build.sh

```diff
@@ -56,6 +56,29 @@ function archive {
     SKIP_INSTALL=NO
 }
 
+function flattenFrameworks {
+  PLATFORM=$1
+
+  for FRAMEWORK in "$ARCHIVES_ROOT/$PLATFORM.xcarchive/Products/Library/Frameworks"/*;
+  do
+    CURRENT_FRAMEWORK_NAME=$(basename "$FRAMEWORK" .framework)
+
+    if [[ -d "$FRAMEWORK/Frameworks" ]]; then
+      log "[$PLATFORM] $CURRENT_FRAMEWORK_NAME has embedded frameworks, flattening them..."
+
+      for EMBEDDED_FRAMEWORK in "$FRAMEWORK/Frameworks"/*;
+      do
+        CURRENT_EMBEDDED_FRAMEWORK_NAME=$(basename "$EMBEDDED_FRAMEWORK" .framework)
+
+        log 'package' "Flattening $CURRENT_EMBEDDED_FRAMEWORK_NAME"
+
+        mv "$EMBEDDED_FRAMEWORK" "$ARCHIVES_ROOT/$PLATFORM.xcarchive/Products/Library/Frameworks"
+      done
+      rm -rf "$FRAMEWORK/Frameworks"
+    fi
+  done
```

Comment:
This function simply moves embedded frameworks (i.e. nested under the `Frameworks` folder within another framework) to the root level.

Reply:
I'm curious if this would be enough to make it work, i.e. I wonder if we don't need to update some `@loadpath` (via some `install_name_tool` dance) so that the dynamic loader (dyld) would be able to properly load the (ex-embded) framework from the parent framework (i.e. load Hermes from Gutenberg) correctly (especially if the load path as printed by `otool -L` would show Hermes being referenced as a relative path from Gutenberg) 🤔 

I _think_ that if you run `otool -L DerivedData/Build/Products/…/WordPress.app/WordPress` on the compiled and linked WordPress executable after it has integrated the `Gutenberg.xcframework` via CocoaPods, you should be able to check that both Gutenberg and Hermes are referenced as `@rpath/[Name].framework/[Name]`? Though I'd be curious to see if `otool -L DerivedData/Build/Products/…/WordPress.app/Frameworks/Gutenberg.framework/Gutenberg` would reference Hermes itself (and if so, how) or not at that point (I'm not familiar with the setup that Gio put in place and how Gutenberg and Hermes interact / might have link-time dependencies on each other)

Reply:
In practice, if we can find out how to do it, I think it would be better to have the archive action generate Hermes and Gutenberg as separate flattened frameworks to begin with at compile time, rather than have Hermes be generated as an embed framework by Xcode in the `xcarchive` it generates, and then have our scripts try to de-embed it manually after the fact… 🤔

Reply:
Thank you so much @AliSoftware for the feedback about this approach 🙇 !

> I'm curious if this would be enough to make it work, i.e. I wonder if we don't need to update some `@loadpath` (via some `install_name_tool` dance) so that the dynamic loader (dyld) would be able to properly load the (ex-embded) framework from the parent framework (i.e. load Hermes from Gutenberg) correctly (especially if the load path as printed by `otool -L` would show Hermes being referenced as a relative path from Gutenberg) 🤔

I'm testing the changes in combination with https://github.com/wordpress-mobile/WordPress-iOS/pull/21021, and so far building the app locally for both the simulator and physical device seems to work. I can take a quick look at the commands you share to validate that references are not broken, although I understand I'd probably experience a build failure or runtime crash in that case 🤔.

> In practice, if we can find out how to do it, I think it would be better to have the archive action generate Hermes and Gutenberg as separate flattened frameworks to begin with at compile time, rather than have Hermes be generated as an embed framework by Xcode in the `xcarchive` it generates, and then have our scripts try to de-embed it manually after the fact… 🤔

Good idea. I assume this would be done by adding a phase script in the Xcode project to flatten the embedded frameworks, is this accurate?

Reply:
> I assume this would be done by adding a phase script in the Xcode project to flatten the embedded frameworks, is this accurate?

I was more thinking about selecting "Do Not Embed" in the "Frameworks and Libraries" settings of the `XCFrameworkScaffold.xcodeproj`:
<img width="824" alt="image" src="https://github.com/wordpress-mobile/gutenberg-mobile/assets/216089/85f3c736-f21b-4e82-a66f-2c919e9080e5">

_Though to be honest, I haven't looked at how the Gutenberg.framework compilation works (how the scaffold project is configured in details and how this interacts with the build scripts like `build.sh` and the declarations in the `podspec` and the potential custom scripts / `post_install` hooks in client app and all that), so I'd prefer to let @mokagio chime in on the details there than trying to grasp at straws without knowing the details about the bigger picture._

Reply:
> I _think_ that if you run `otool -L DerivedData/Build/Products/…/WordPress.app/WordPress` on the compiled and linked WordPress executable after it has integrated the `Gutenberg.xcframework` via CocoaPods, you should be able to check that both Gutenberg and Hermes are referenced as `@rpath/[Name].framework/[Name]`? Though I'd be curious to see if `otool -L DerivedData/Build/Products/…/WordPress.app/Frameworks/Gutenberg.framework/Gutenberg` would reference Hermes itself (and if so, how) or not at that point (I'm not familiar with the setup that Gio put in place and how Gutenberg and Hermes interact / might have link-time dependencies on each other)

I've run the commands and seems the reference is correct. For both the app's executable and Gutenberg framwork, they point to Hermes (`@rpath/hermes.framework`).


Result of `otool -L DerivedData/.../Build/Products/Debug-iphoneos/Jetpack.app/Jetpack`:

```
[...]
@rpath/Gutenberg.framework/Gutenberg (compatibility version 1.0.0, current version 1.0.0)
@rpath/hermes.framework/hermes (compatibility version 0.12.0, current version 0.12.0)
[...]
```

Result of `otool -L DerivedData/.../Build/Products/Debug-iphoneos/Jetpack.app/Frameworks/Gutenberg.framework/Gutenberg`:

```
[...]
@rpath/Gutenberg.framework/Gutenberg (compatibility version 1.0.0, current version 1.0.0)
@rpath/hermes.framework/hermes (compatibility version 0.12.0, current version 0.12.0)
[...]
```

Reply:
> I was more thinking about selecting "Do Not Embed" in the "Frameworks and Libraries" settings of the XCFrameworkScaffold.xcodeproj:

FWIW, #5973 uses that approach.